executor/write_test.go

```diff
@@ -1389,6 +1390,7 @@ func (s *testSuite8) TestUpdate(c *C) {
 
 	// Test that in a transaction, when a constraint failed in an update statement, the record is not inserted.
 	tk.MustExec("create table update_unique (id int primary key, name int unique)")
+	tk.MustExec("set @@tidb_constraint_check_in_place = 1")
```

Comment:
Why do we need to change this test?

Reply:
There is no `begin` keyword before the `INSERT` statement, Is it running in auto-commit mode.

Reply:
auto-commit statements should return the same error or no error no mater `tidb_constraint_check_in_place` is set to '0' or '1'.